### PR TITLE
Refactor "Devices" tab from List to Table from SwiftUI

### LINKED ISSUE #54: Refactor `ParametersTable` component to `Table`
Currently `ParametersTable` uses "custom" implementation of Table due to MacOS 11 as a target. However, with 0.6.0 release MiniSim will support only MacOS 12+ so this component can be refactored.

### PR DESCRIPTION
**Pull request by jaworek**
Resolves #54.

Video recording showing new implementation:

https://github.com/okwasniewski/MiniSim/assets/22678765/cba0920c-8a3b-46ee-9953-96e8c46ee0b7

### CODE REVIEW

File: MiniSim/Views/ParametersTable/ParametersTableViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                return
+            }
+            guard var parameter = parameters.first(where: { $0.id == item }) else {
+                return
+            }
+            parameters.removeAll(where: { $0.id == item })
+            parameter.enabled = !parameter.enabled
+            parameters.insert(parameter, at: index)
+            saveData()
+        }
```
Comment (jaworek) [inline MiniSim/Views/ParametersTable/ParametersTableViewModel.swift:48]: This implementation seems a bit messy. I'm open to suggestions on how to handle it better :)

File: MiniSim/Views/ParametersTable/ParametersTable.swift [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+//
+//  Created by Oskar Kwasniewski on 12/03/2023.
+//
+
+import SwiftUI
+
+struct ParametersTable: View {
+    @State private var parameters: [Parameter] = [.init(title: "", command: "", enabled: false)]
+    
+    func saveData() {
```
Comment (okwasniewski) [inline MiniSim/Views/ParametersTable/ParametersTable.swift:13]: This code is duplicated from the ViewModel

File: MiniSim/Views/ParametersTable/ParametersTable.swift [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                }
+            }
+
+            HStack {
+                Spacer()
+                Button("Add New") {
+                    viewModel.selection = nil
+                    viewModel.showForm.toggle()
+                }.buttonStyle(.borderedProminent)
+                    .keyboardShortcut("n", modifiers: .command)
```
Comment (okwasniewski) [inline MiniSim/Views/ParametersTable/ParametersTable.swift:70]: Something is off with the formatting here, can you keep each modifier on new line?

File: MiniSim/Views/ParametersTable/ParametersTableForm.swift [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                    source: $viewModel.command,
+                    language: .bash,
+                    theme: colorScheme == .dark ? .atelierSavannaDark : .atelierSavannaLight,
+                    flags: [.selectable, .editable, .smartIndent]
+                )
+                .cornerRadius(6)
+            }
+            Toggle(isOn: $viewModel.enabled, label: {
+                Text("Enabled")
+            })
```
Comment (okwasniewski) [inline MiniSim/Views/ParametersTable/ParametersTableForm.swift:37]: Looks like this doesn't need to accept custom label:

```swift
Toggle("Enabled", isOn: $viewModel.enabled)

File: MiniSim/Views/ParametersTable/ParametersTableViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        
+        func loadData() {
+            guard let paramData = UserDefaults.standard.parameters else { return }
+            if let decodedData = try? JSONDecoder().decode([Parameter].self, from: paramData) {
+                parameters = decodedData
+            }
+        }
+        
+        func deleteParameters(item: Parameter.ID?) {
```
Comment (okwasniewski) [inline MiniSim/Views/ParametersTable/ParametersTableViewModel.swift:32]: This is deleting one parameter so you can rename this to `deleteParameter`

File: MiniSim/Views/ParametersTable/ParametersTableViewModel.swift (hunk 1 above)
Comment (okwasniewski) [inline MiniSim/Views/ParametersTable/ParametersTableViewModel.swift:48]: I would change it as follows: 

```swift
func toggleEnabled(item: Parameter.ID?) {
            guard let index = parameters.firstIndex(where: { $0.id == item }) else {
                return
            }
            var paremeter = parameters.remove(at: index)
            paremeter.enabled.toggle() // or paremeter.enabled = !item.enabled
            paremeter.insert(parameter, at: index)
            saveData()
        }
```

**Review by okwasniewski — COMMENTED** [review]
Thanks for contributing! I've left few minor comments

**Comment by jaworek** [comment]
Thanks for the quick review! I've addressed all suggestions :D

File: MiniSim/Views/ParametersTable/ParametersTable.swift [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+//  ParametersTable.swift
+//  MiniSim
+//
+//  Created by Oskar Kwasniewski on 12/03/2023.
+//
+
+import SwiftUI
+
+struct ParametersTable: View {
+    @State private var parameters: [Parameter] = [.init(title: "", command: "", enabled: false)]
```
Comment (okwasniewski) [inline MiniSim/Views/ParametersTable/ParametersTable.swift:11]: Looks like this is not needed anymore

File: MiniSim/Views/ParametersTable/ParametersTableForm.swift [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+                    viewModel.clearForm()
+                }
+                .buttonStyle(.borderedProminent)
+                .disabled(viewModel.disableForm)
+                .keyboardShortcut("s", modifiers: .command)
+            }
+            .padding(.top)
+        }
+        .padding()
+        .frame(minWidth: 550, minHeight: 400)
```
Comment (okwasniewski) [inline MiniSim/Views/ParametersTable/ParametersTableForm.swift:65]: Let's change the height to 250 as user is expected to only input the parameter like (`--no-audio`)

**Review by okwasniewski — COMMENTED** [review]
When I launched the example files weren't added to Xcode (they were on disk) did you push the changes in `project.pbxproj`?

**Review by okwasniewski — APPROVED** [review]
Great work, thanks for this! 🙏

**Pull request merged**